static/skin/index.js

```diff
@@ -1,62 +1,68 @@
+    const title =  getInnerHtml(book, 'title');
+    const description = getInnerHtml(book, 'summary');
+    const id = getInnerHtml(book, 'id');
+    const iconUrl = getInnerHtml(book, 'icon');
+    const articleCount = getInnerHtml(book, 'articleCount');
+    const mediaCount = getInnerHtml(book, 'mediaCount');
+
+    return `<a href='${link}' data-id='${id}'><div class='book'>
+        <div class='book__background' style="background-image: url('${iconUrl}');">
+        <div class='book__title' title='${title}'>${title}</div>
+        <div class='book__description' title='${description}'>${description}</div>
+        <div class='book__info'>${articleCount} articles, ${mediaCount} medias</div>
+        </div>
+    </div></a>`;
+}
+
+async function loadAndDisplayBooks() {
     isFetching = true;
     fetch(queryUrlBuilder()).then(async (resp) => {
         const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
         const books = data.querySelectorAll('entry');
         let bookHtml = '';
-        books.forEach((book) => {
-            const link = book.querySelector('link').getAttribute('href');
-            const title =  getInnerHtml(book, 'title');
-            const description = getInnerHtml(book, 'summary');
-
-            bookHtml += `<a href='${link}' data-id='${getInnerHtml(book, 'id')}'><div class='book'>
-                <div class='book__background' style="background-image: url('${getInnerHtml(book, 'icon')}');">
-                <div class='book__title' title='${title}'>${title}</div>
-                <div class='book__description' title='${description}'>${description}</div>
-                <div class='book__info'>${getInnerHtml(book, 'articleCount')} articles, ${getInnerHtml(book, 'mediaCount')} medias</div>
-                </div>
-            </div></a>`;
-        });
-        document.querySelector('.book__list').innerHTML = (append ? document.querySelector('.book__list').innerHTML : '') + bookHtml;
-        isEnd = !books.length;
+        books.forEach((book) => {bookHtml += generateBookHtml(book)});
+        document.querySelector('.book__list').innerHTML += bookHtml;
+        incrementalLoadingParams['start'] = incrementalLoadingParams['start'] + incrementalLoadingParams['count'];
+        if (!books.length) {
+            incrementalLoadingParams['count'] = 0;
+        }
+        isFetching = false;
     });
-    isFetching = false;
 }
 
 window.addEventListener('resize', async () => {
-    if (isFetching || isEnd) return;
-    viewPortHeight = window.innerHeight;
-    basicConfig['start'] = basicConfig['start'] + basicConfig['count'];
-    basicConfig['count'] = Math.floor(viewPortHeight/100 + 1) * 3;
-    loadAndDisplayBooks(true);
+    if (isFetching || !incrementalLoadingParams['count'] || incrementalLoadingParams['count'] > viewPortToCount()) return;
+    incrementalLoadingParams['count'] = incrementalLoadingParams['count'] && viewPortToCount();
+    loadAndDisplayBooks();
```

Comment:
We need to load more books only if we are at the end of the book list after the resize.